File: client/android/div-core-views/src/main/java/com/yandex/div/view/pooling/ViewPool.java
package com.yandex.div.view.pooling;

import android.view.View;
import androidx.annotation.NonNull;

public interface ViewPool {

    <T extends View> void register(@NonNull String tag, @NonNull ViewFactory<T> factory, int capacity);

    void unregister(@NonNull String tag);

    @NonNull
    <T extends View> T obtain(@NonNull String tag);
}

File: client/android/div/src/main/java/com/yandex/div/internal/viewpool/ViewPool.kt
package com.yandex.div.internal.viewpool

import android.view.View

interface ViewPool {

    fun <T : View> register(tag: String, factory: ViewFactory<T>, capacity: Int)

    fun unregister(tag: String)

    fun <T : View> obtain(tag: String): T

    fun changeCapacity(tag: String, newCapacity: Int)
}
